### PR TITLE
go.{mod,sum}, .github: bump minimum golang version to 1.20.

### PR DESCRIPTION
**Pull request by klihub**
- bump minimum golang version to 1.20
- test using 1.20, 1.21, and 1.22 in CI
- bump setup-go to v5
- update golangci-lint and fix related linter errors

### CODE REVIEW

**Review by fuweid — APPROVED** [review]

**Review by Iceber — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
 
     strategy:
       matrix:
-        go: [1.19.x, 1.20.x]
+        go: [1.20.x]
```
Comment (mikebrow) [inline .github/workflows/ci.yml:63]: should we also check 1.21 and/or 1.22.4?  noting we moved to actions/setup-go@v5 in core..

Comment (klihub) [reply .github/workflows/ci.yml:64]: Okay, added another commit to bump setup-go to v5, too.

Comment (thaJeztah) [reply .github/workflows/ci.yml:63]: Agreed, I think it makes sense to _either_ test min (1.20) and latest (1.22), or if CI is short to test all of 1.20, 1.21, 1.22.

File: plugins/network-device-injector/go.mod [hunk 2]
```diff
@@ -2,6 +2,8 @@ module github.com/containerd/nri/plugins/network-device-injector
 
 go 1.22.0
 
+toolchain go1.22.1
```
Comment (mikebrow) [inline plugins/network-device-injector/go.mod:5]: interesting :)

**Review by mikebrow — APPROVED** [review]
LGTM on green

File: plugins/network-device-injector/go.mod (hunk 2 above)
Comment (klihub) [reply plugins/network-device-injector/go.mod:5]: Hmm... my bad, was unintentional. Removed.

**Review by mikebrow — APPROVED** [review]
LGTM

**Pull request merged**